src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,364 @@
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    /// <typeparam name="T">A user-defined POCO that represents a row of the user's table</typeparam>
+    internal sealed class SqlTriggerListener<T> : IListener
+    {
+        private readonly string _userTableName;
+        private readonly string _connectionString;
+        private readonly string _userFunctionId;
+        private readonly ITriggeredFunctionExecutor _executor;
+        private readonly ILogger _logger;
+
+        private SqlTableChangeMonitor<T> _changeMonitor;
+        private State _state;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqlTriggerListener{T}" />>
+        /// </summary>
+        /// <param name="connectionString">The SQL connection string used to connect to the user's database</param>
+        /// <param name="tableName">The name of the user table whose changes are being tracked on</param>
+        /// <param name="userFunctionId">
+        /// The unique ID that identifies user function. If multiple application instances are executing the same user
+        /// function, they are all supposed to have the same user function ID.
+        /// </param>
+        /// <param name="executor">Used to execute the user's function when changes are detected on "table"</param>
+        /// <param name="logger">Ilogger used to log information and warnings</param>
+        public SqlTriggerListener(string connectionString, string tableName, string userFunctionId, ITriggeredFunctionExecutor executor, ILogger logger)
+        {
+            _ = !string.IsNullOrEmpty(connectionString) ? true : throw new ArgumentNullException(nameof(connectionString));
+            _ = !string.IsNullOrEmpty(tableName) ? true : throw new ArgumentNullException(nameof(tableName));
+            _ = !string.IsNullOrEmpty(userFunctionId) ? true : throw new ArgumentNullException(nameof(userFunctionId));
+            _ = executor ?? throw new ArgumentNullException(nameof(executor));
+            _ = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            this._connectionString = connectionString;
+            this._userTableName = new SqlObject(tableName).FullName;
+            this._userFunctionId = userFunctionId;
+            this._executor = executor;
+            this._logger = logger;
+            this._state = State.NotInitialized;
+        }
+
+        /// <summary>
+        /// Stops the listener which stops checking for changes on the user's table.
+        /// </summary>
+        public void Cancel()
+        {
+            this.StopAsync(CancellationToken.None).Wait();
+        }
+
+        /// <summary>
+        /// Disposes resources held by the listener to poll for changes.
+        /// </summary>
+        public void Dispose()
+        {
+            // Nothing to dispose.
```

Comment:
Shouldn't you cancel the listener when this happens?

Reply:
The WebJobs SDK calls either `StopAsync` or `Cancel` on the `SqlTriggerListener` which in turn stops the change-monitor. I can see extensions like RabbitMQ having empty `Dispose` method while [EventHub calling `StopAsync` from inside both `Cancel` and `Dispose`](https://github.com/Azure/azure-sdk-for-net/blob/a5f2ba22c34a6d931b3717aa36f1c286c4647d13/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs#L65-L95). I will update the code following this same pattern.